app/factories/bulkrax/object_factory.rb

```diff
@@ -182,12 +182,20 @@ def update_file_set(attrs)
       actor.update_metadata(file_set_attrs)
     end
 
+    def pass_collection_to_be_persisted(parent:, child:)
```

Comment:
I created this method because the `parent` that gets returned from the `find_collection` method down below has always been an array.

I'm not sure when it will be a hash or a string (except when an array of strings is being mapped over), but the options are there so I didn't wanna discount them.

@orangewolf @sephirothkod @bkiahstroud have any of you recently seen where the id being passed into the `find_collection` method is a hash or string? do we still need to account for "when Hash"?

Reply:
> I'm not sure when it will be a hash or a string (except when an array of strings is being mapped over) 

For this reason, I'm confident that both Array and String are valid here. 

As far as Hash goes, I've never seen this used. Personally, I'd remove it unless someone knows for sure if it gets used by something I'm not super familiar with (i.e. the OAI or Bagit parsers). If we need to handle Hashes in the future, we can always add it back in.

Reply:
I feel like I remember bagit having id's in hashes as a possibility, let's keep accounting for this.

Reply:
Whoops, I didn't mean to request a change here. I meant it to be for the call being moved to after the object.save sorry for the confusion!